src/plugins/intel_npu/src/plugin/npuw/lazy_tensor.cpp

```diff
@@ -4,42 +4,55 @@
 
 #include "lazy_tensor.hpp"
 
-using ov::npuw::weights::ConcatMeta;
-using ov::npuw::weights::ConstPtr;
 using ov::npuw::weights::LazyTensor;
-using ov::npuw::weights::OrigData;
-using ov::npuw::weights::Transform;
-using ov::npuw::weights::TransformType;
-using ov::npuw::weights::UnpackMeta;
 
 namespace ov {
 namespace npuw {
 namespace weights {
 
+enum class TransformType : int { TENSOR, CONST, CONCAT, UNPACK, PERMUTE, CONVERT };
+using ConcatMeta = std::pair<std::vector<LazyTensor>, std::size_t>;
+using UnpackMeta = std::tuple<LazyTensor, LazyTensor, LazyTensor, ov::Shape, ov::element::Type>;
+using ConstPtr = std::shared_ptr<ov::op::v0::Constant>;
```

Comment:
Let's make these distinct types with meaningful fields. Like

```suggestion
namespace op {
    struct Const {
        std::shared_ptr<ov::op::v0::Constant> node;
    };
    struct Unpack {
        LazyTensor w, z, s;
        ov::element::Type dst_type;
        ov::Shape dst_shape; // BTW, why would you need this?
    };
    struct Concat {
        std::vector<LazyTensor> tensors;
        std::size_t axis;
    };
}
```
Let's agree here that each `op::` class here carries in operation _AND_ data to apply this operation on.
So if you add `ToF16`, there must be its lazy tensor operand ("what to convert") in the struct. If you add `Permute`, its lazy tensor operand ("what to permute") should be there as well.

Reply:
Done